zerver/lib/cache.py

```diff
@@ -103,6 +107,23 @@ def get_cache_backend(cache_name):
         return djcache
     return get_cache(cache_name)
 
+def get_cache_with_key(keyfunc, cache_name=None):
+    # type: (Any, Optional[str]) -> Any
+    def decorator(func):
+        # type: (Callable[..., Any]) -> (Callable[..., Any])
+        @wraps(func)
+        def func_with_caching(*args, **kwargs):
+            # type: (*Any, **Any) -> Callable[..., Any]
+            key = keyfunc(*args, **kwargs)
+            val = cache_get(key, cache_name=cache_name)
+            if val is not None:
+                return val[0]
+            raise NotFoundInCache()
+
+        return func_with_caching
+
+    return decorator
```

Comment:
Why do you need this new caching function -- is the idea that you want to check if it's in the cache but not go to the original source if it's not present in the cache, instead raising an exception?  

I think it's worth giving this a clearer name and a docstring similar to that of `cache_with_key`.